ClientUpdater/Updater.cs

```diff
@@ -15,1140 +15,1152 @@
+    /// <param name="archiveSize">Archive file size.</param>
+    internal static UpdaterFileInfo CreateFileInfo(string filename, string identifier, int size, string archiveIdentifier = null, int archiveSize = 0)
+    {
+        return new()
+        {
+            Filename = SafePath.CombineFilePath(filename),
+            Identifier = identifier,
+            Size = size,
+            ArchiveIdentifier = archiveIdentifier,
+            ArchiveSize = archiveSize
+        };
+    }
 
-        #endregion
+    internal static void UpdateUserAgent(HttpClient httpClient)
+    {
+        httpClient.DefaultRequestHeaders.UserAgent.Clear();
+        httpClient.DefaultRequestHeaders.UserAgent.Add(new(LocalGame, GameVersion));
 
-        #region internal_methods
+        if (UpdaterVersion != "N/A")
+            httpClient.DefaultRequestHeaders.UserAgent.Add(new("Updater", UpdaterVersion));
 
-        /// <summary>
-        /// Get archive info for a file from version file.
-        /// </summary>
-        /// <param name="versionFile">Version file.</param>
-        /// <param name="filename">Filename.</param>
-        /// <param name="archiveID">Set to archive ID.</param>
-        /// <param name="archiveSize">Set to archive file size.</param>
-        internal static void GetArchiveInfo(IniFile versionFile, string filename, out string archiveID, out int archiveSize)
-        {
-            string[] values = versionFile.GetStringValue("ArchivedFiles", filename, "").Split(',');
-            bool archiveAvailable = values != null && values.Length >= 2;
-            archiveID = archiveAvailable ? values[0] : "";
-            archiveSize = archiveAvailable ? Conversions.IntFromString(values[1], 0) : 0;
-        }
+        httpClient.DefaultRequestHeaders.UserAgent.Add(new("Client", Assembly.GetEntryAssembly().GetName().Version.ToString()));
+    }
 
-        /// <summary>
-        /// Creates file info instance from given information.
-        /// </summary>
-        /// <param name="filename">Filename.</param>
-        /// <param name="identifier">File identifier.</param>
-        /// <param name="size">File size.</param>
-        /// <param name="archiveIdentifier">Archive file identifier.</param>
-        /// <param name="archiveSize">Archive file size.</param>
-        /// <returns></returns>
-        internal static UpdaterFileInfo CreateFileInfo(string filename, string identifier, int size, string archiveIdentifier = null, int archiveSize = 0)
+    /// <summary>
+    /// Deletes file and waits until it has been deleted.
+    /// </summary>
+    /// <param name="filepath">File to delete.</param>
+    /// <param name="timeout">Maximum time to wait in milliseconds.</param>
+    internal static void DeleteFileAndWait(string filepath, int timeout = 10000)
+    {
+        FileInfo fileInfo = SafePath.GetFile(filepath);
+        using (var fw = new FileSystemWatcher(fileInfo.DirectoryName, fileInfo.Name))
+        using (var mre = new ManualResetEventSlim())
```

Comment:
Why not use just `using var ...` instead of this quite unconventional construct?